cardano-db/src/Cardano/Db/Operations/Query.hs

```diff
@@ -218,22 +219,39 @@ queryBlockNoAndEpoch blkNo = do
     blk <- from $ table @Block
     where_ (blk ^. BlockBlockNo ==. just (val blkNo))
     pure (blk ^. BlockId, blk ^. BlockEpochNo)
-  pure $ convert (listToMaybe res)
-  where
-    convert :: Maybe (Value (Key Block), Value (Maybe Word64)) -> Maybe (BlockId, Word64)
-    convert mr =
-      case mr of
-        Nothing -> Nothing
-        Just (_, Value Nothing) -> Nothing -- Should not ever happen.
-        Just (Value blkid, Value (Just epoch)) -> Just (blkid, epoch)
-
-queryBlockSlotNo :: MonadIO m => Word64 -> ReaderT SqlBackend m (Maybe BlockId)
+  pure $ convertBlockQuery (listToMaybe res)
+
+queryBlockSlotNo :: MonadIO m => Word64 -> ReaderT SqlBackend m (Maybe (BlockId, Word64))
 queryBlockSlotNo slotNo = do
   res <- select $ do
     blk <- from $ table @Block
     where_ (blk ^. BlockSlotNo ==. just (val slotNo))
```

Comment:
Could we change the behaviour here, so that it deletes all blocks with greater or equal slot_no?
Currently it finds blocks with the provided slot_no and only if there is one it deletes it and everythiong after it. This is a bit counter-intuitive.